src/test/util_sds_check/util_sds_check.c

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2025, Hewlett Packard Enterprise Development LP */
+
+/*
+ * util_sds_check.c -- unit test for the shutdown_state_check() function
+ */
+
+#include "util_sds_check.h"
+
+/* pool_sds - SDS read from the pool header (persistent) */
+static struct shutdown_state Pool_sds;
+/*
+ * curr_sds - SDS calculated ad hoc considering all the involved PMem DIMMs
+ * their UUIDs and USCs (volatile)
+ */
+static struct shutdown_state Curr_sds;
+static struct pool_replica Rep;
+
+FUNC_MOCK(shutdown_state_reinit, void, struct shutdown_state *curr_sds,
+	struct shutdown_state *pool_sds, struct pool_replica *rep)
+FUNC_MOCK_RUN_DEFAULT {
+	UT_ASSERTeq(curr_sds, &Curr_sds);
+	UT_ASSERTeq(pool_sds, &Pool_sds);
+	UT_ASSERTeq(rep, &Rep);
+}
+FUNC_MOCK_END
+
+/*
+ * The persistent SDS turned out to be zeroed out
+ * whereas based on the pool structure it should be not.
+ */
+static int
+test_dirty_clear()
+{
+	SET_SDS(Pool_sds, SDS_ZERO);
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
+
+	int ret = shutdown_state_check(&Curr_sds, &Pool_sds, &Rep);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(RCOUNTER(shutdown_state_reinit), 1);
+
+	return ret;
+}
+
+/* Pool_sds having an invalid checksum. */
+static int
+test_invalid_checksum()
+{
+	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
+	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
+	shutdown_state_checksum(&Curr_sds, &Rep);
```

Comment:
1. Adding a comment here would be nice.
2. `Curr_sds.checksum` is irrelevant.

```suggestion
	SET_SDS(Pool_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
	/* Pool_sds.checksum left zeroed out (invalid) */
	SET_SDS(Curr_sds, SDS_CUSTOM(USC, UUID, DIRTY_CLEAR));
```

Reply:
Done.